src/modules/navigator/land.cpp

```diff
@@ -93,4 +93,20 @@ Land::on_active()
 		mission_item_to_position_setpoint(_mission_item, &pos_sp_triplet->current);
 		_navigator->set_position_setpoint_triplet_updated();
 	}
+
+	/* check if landing needs to be aborted */
+	if (_navigator->abort_landing()) {
+
+		// send reposition cmd to get out of land mode (will loiter at current position and altitude)
```

Comment:
i think i already made a comment on this somewhere else in the thread - but just to be sure - are you sure you don't want the minimum altitude set on this? otherwise could abort like 5m above the ground.

Reply:
The thing is that we have no way of knowing the altitude AMSL of the land point, so I would keep it as simple as possible. For the mission landing the land point AMSL is ofc known, so there we can have a more extensive abort logic.

Reply:
just seems a bit dangerous to me. would it make sense to e.g. store the starting altitude and climb back up there?

Reply:
>store the starting altitude and climb back up there

What I don't like with that is that the behavior strongly diverges from the mission landing abort, plus when triggering the land from high up then the vehicle would change its position a lot after the "abort" message was sent. And that's not what I would expect of an "abort". 
I really also want to make this as simple as possible, and would sacrifice some corner case UX deficits for a very predictable outcome and light logic to maintain. Especially until we get real-world experience in.

Reply:
Alright after discussing in yesterday's call and giving it more though:
- manual abort is enabled
- automatic abort based on terrain not found is disabled (as we don't know when to expect valid terrain for the circular landing)
- automatic abort based on terrain estimation timeout is enabled
- respect MIS_LND_ABRT_ALT also for landings outside of missions, and assume worst case (we're right at terrain altitude when triggering the abort) --> always climb MIS_LND_ABRT_ALT meters on triggering the abort.

Reply:
Awesome! Will give this a SITL try.